### PR TITLE
Add support for backups

### PR DESCRIPTION
**Pull request by alexbakker**
Allow users to select a folder where automatic backups will be created.

### CODE REVIEW

**Comment by michaelschattgen** [comment]
Nice, great feature! However, to keep things secure I suggest to currently disable this feature when the user doesn't use encryption. I think we have another issue where seperate encryption for backups is suggested.

**Comment by alexbakker** [comment]
@michaelschattgen

> However, to keep things secure I suggest to currently disable this feature when the user doesn't use encryption.

Agreed. I'll change that.

> I think we have another issue where seperate encryption for backups is suggested.

You're thinking of #121. So far I've only thought about how to implement that for encrypted vaults (we'd just add an extra key slot). I'm not sure what the best way to implement that for unencrypted vaults would be, but we can discuss that later.

**Comment by alexbakker** [comment]
Done.

**Comment by alexbakker** [comment]
This now also bumps minSdkVersion to 21.

File: app/src/main/java/com/beemdevelopment/aegis/ui/Dialogs.java [hunk 1]
```diff
@@ -214,6 +214,31 @@ public static void showNumberPickerDialog(Activity activity, NumberInputListener
         showSecureDialog(dialog);
     }
 
+    public static void showBackupVersionsPickerDialog(Activity activity, NumberInputListener listener) {
+        final int max = 30;
+        String[] numbers = new String[30 / 5];
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/ui/Dialogs.java:219]: You forgot to use the variable 'max' here.

File: app/src/main/java/com/beemdevelopment/aegis/util/IOUtils.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class IOUtils {
+    private IOUtils() {
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/util/IOUtils.java:11]: Empty constructor, looks like it can be removed.

Comment (alexbakker) [reply app/src/main/java/com/beemdevelopment/aegis/util/IOUtils.java:11]: This private constructor is there to that the class can't be instantiated, as it only contains static methods.

Comment (michaelschattgen) [reply app/src/main/java/com/beemdevelopment/aegis/util/IOUtils.java:11]: TIL Java doesn't support static classes.

**Pull request merged**